pywbemtools/pywbemcli/_utils.py

```diff
@@ -108,14 +109,50 @@ def _eq_name(name1, name2):
     return name1.lower() == name2.lower()
 
 
-def formatwarning(message, category, filename, lineno, line=None):
+def _formatwarning(message, category, filename, lineno, line=None):
     # pylint: disable=unused-argument
     """
-    Replacement for warnings.formatwarning() that is patched in.
+    Replacement for warnings.formatwarning() that replaces the standard
+    warning;.formatwarning for pywbemcli.
     """
     return "{}: {}\n".format(category.__name__, message)
 
 
+def pywbemcliwarn(*args, **kwargs):
+    """
+    Pywbemcli replacement for the warnings.warn function that substitutes our
+    own formatter.
+    """
+    # Import for each call
+    # pylint: disable=import-outside-toplevel, cyclic-import
+    from .pywbemcli import WARN_OPTION
+    if WARN_OPTION:
+        warnings.simplefilter('once')
```

Comment:
Which problem is solved by moving the `warnings.simplefilter()` call to here? It introduces the need for the global `WARN_OPTION` so let's do that only if there is a good reason for it.

Reply:
I defy you to find out exactly what  the scope of the filters in warnings is from the documentation.  I am still not sure but it appears that it applies across either a context in which warnings is applied or the complete package and the multiple "import warnings" in pywbemcli.py and _utils.py are irrelevant. O learn.

Removed the global WARN_OPTION, and the setting of the WARN_OPTION in _utils.py.